prowler/providers/aws/aws_provider_new.py

```diff
@@ -0,0 +1,622 @@
+from prowler.providers.aws.config import (
+    AWS_STS_GLOBAL_ENDPOINT_REGION,
+    BOTO3_USER_AGENT_EXTRA,
+)
+from prowler.providers.aws.lib.arn.arn import parse_iam_credentials_arn
+from prowler.providers.aws.lib.credentials.credentials import (
+    create_sts_session,
+    validate_aws_credentials,
+)
+from prowler.providers.aws.lib.organizations.organizations import (
+    get_organizations_metadata,
+)
+from prowler.providers.common.provider import CloudProvider
+
+
+@dataclass
+class AWS_Organizations_Info:
+    account_details_email: str
+    account_details_name: str
+    account_details_arn: str
+    account_details_org: str
+    account_details_tags: str
+
+
+@dataclass
+class AWS_Credentials:
+    aws_access_key_id: str
+    aws_session_token: str
+    aws_secret_access_key: str
+    expiration: datetime
+
+
+@dataclass
+class AWS_Assume_Role:
+    role_arn: str
+    session_duration: int
+    external_id: str
+    mfa_enabled: bool
+
+
+@dataclass
+class AWSAssumeRoleConfiguration:
+    assumed_role_info: AWS_Assume_Role
+    assumed_role_credentials: AWS_Credentials
+
+
+@dataclass
+class AWSIdentityInfo:
+    account: str
+    account_arn: str
+    user_id: str
+    partition: str
+    identity_arn: str
+    profile: str
+    profile_region: str
+    audited_regions: list
+
+
+@dataclass
+class AWSSession:
+    session: session.Session
+    session_config: Config
+    original_session: None
+
+
+class AwsProvider(CloudProvider):
+    session: AWSSession = AWSSession(
+        session=None, session_config=None, original_session=None
+    )
+    identity: AWSIdentityInfo = AWSIdentityInfo(
+        account=None,
+        account_arn=None,
+        user_id=None,
+        partition=None,
+        identity_arn=None,
+        profile=None,
+        profile_region=None,
+        audited_regions=[],
+    )
+    assumed_role: AWSAssumeRoleConfiguration = AWSAssumeRoleConfiguration(
+        assumed_role_info=AWS_Assume_Role(
+            role_arn=None,
+            session_duration=None,
+            external_id=None,
+            mfa_enabled=False,
+        ),
+        assumed_role_credentials=AWS_Credentials(
+            aws_access_key_id=None,
+            aws_session_token=None,
+            aws_secret_access_key=None,
+            expiration=None,
+        ),
+    )
+    organizations_metadata: AWS_Organizations_Info = AWS_Organizations_Info(
+        account_details_email=None,
+        account_details_name=None,
+        account_details_arn=None,
+        account_details_org=None,
+        account_details_tags=None,
+    )
+    audit_resources: Optional[Any] = None
+    audit_metadata: Optional[Any] = None
+    audit_config: dict = {}
+    mfa_enabled: bool = False
+    ignore_unused_services: bool = False
+
+    def __init__(self, args):
+        logger.info("Setting AWS provider ...")
+        # Parse input arguments
+        # Assume Role Options
+        arguments = args.__dict__
```

Comment:
I think you don't need that if you are receiving the `args` as an argparse `Namespace`.

Also all the `get()` below can be changed, e.g.: `arguments.role`.

Reply:
done